src/System.Security.Cryptography.Cng/src/System/Security/Cryptography/ECDsaCng.ImportExport.cs

```diff
@@ -0,0 +1,558 @@
+                //     BCRYPT_ECCKEY_BLOB   header
+                //     byte[cbKey]          Qx
+                //     byte[cbKey]          Qy
+                //     -- Only if "includePrivate" is true --
+                //     byte[cbKey]          D
+                //
+
+                int blobSize = sizeof(BCRYPT_ECCKEY_BLOB) +
+                    parameters.Q.X.Length + 
+                    parameters.Q.Y.Length;
+                if (includePrivate)
+                {
+                    blobSize += parameters.D.Length;
+                }
+
+                byte[] blob = new byte[blobSize];
+                fixed (byte* pBlob = blob)
+                {
+                    // Build the header
+                    BCRYPT_ECCKEY_BLOB* pBcryptBlob = (BCRYPT_ECCKEY_BLOB*)pBlob;
+                    pBcryptBlob->Magic = CurveNameToMagicNumber(curve.Oid.FriendlyName, includePrivate);
+                    pBcryptBlob->cbKey = parameters.Q.X.Length;
+
+                    // Emit the blob
+                    int offset = sizeof(BCRYPT_ECCKEY_BLOB);
+                    Emit(blob, ref offset, parameters.Q.X);
+                    Emit(blob, ref offset, parameters.Q.Y);
+                    if (includePrivate)
+                    {
+                        Emit(blob, ref offset, parameters.D);
+                    }
+
+                    // We better have computed the right allocation size above!
+                    Debug.Assert(offset == blobSize, "offset == blobSize");
+                }
+
+                return ImportKeyBlob(blob, curve, includePrivate);
+            }
+        }
+    
+        private static CngKey ImportPrimeCurveParameters(ECParameters parameters, ECPrimeCurve curve, bool includePrivate)
+        {
+            unsafe
+            {
+                //
+                // We need to build a key blob structured as follows:
+                //     BCRYPT_ECCFULLKEY_BLOB       header
+                //     byte[cbFieldLength]          P
+                //     byte[cbFieldLength]          A
+                //     byte[cbFieldLength]          B
+                //     byte[cbFieldLength]          Gx
+                //     byte[cbFieldLength]          Gy
+                //     byte[cbSubgroupOrder]        N
+                //     byte[cbCofactor]             H
+                //     byte[cbSeed]                 S
+                //     byte[cbFieldLength]          Qx
+                //     byte[cbFieldLength]          Qy
+                //     -- Only if "includePrivate" is true --
+                //     byte[cbSubgroupOrder]        D
+
+                int blobSize = sizeof(BCRYPT_ECCFULLKEY_BLOB) +
+                    curve.P.Length +
+                    curve.A.Length +
+                    curve.B.Length +
+                    curve.G.X.Length +
+                    curve.G.Y.Length +
+                    curve.N.Length +
+                    curve.H.Length +
+                    curve.S.Length +
```

Comment:
Currently export will always new up a byte[0], so these won't be null on export and then import. However I see your point as that is not obvious for cases where the parameters are explicitly set without exporting first, so I will change to allow null. That reminds me of a discussion topic w.r.t. padding the beginning of the byte[] when the leading values are zero - should we allow that for cng, as for openssl that is the default using bignum meaning for example an X and Y may have different byte counts in openssl if there are some leading zeros.